app/views/home/generate.php

```diff
@@ -38,7 +38,7 @@
                                 <div class="form-row">
                                     <div class="form-group col-md-6">
                                         <label class="small text-gray-600" for="made_in">Fait à</label>
-                                        <input class="form-control rounded-pill" id="made_in" name="made_in" type="text" placeholder="Ville" value="<?= $made_in ?>" />
+                                        <input class="form-control rounded-pill" id="made_in" name="made_in" type="text" placeholder="Ville" value="<?= $made_in ?>" required />

```

Comment:
Adding the `required` attribute in HTML won't prevent any PHP error (never trust user input). `made_in` HTML field may be empty and it shouldn't be an issue, because a default value is set in the user data. Plus, the presence of the "made_in" argument is checked on back-end side.

But actually, there is an error (which doesn't come from PHP) when the argument is empty. If you want to prevent it, you can test the emptiness of the argument on line 31 of `/app/controllers/attestation.php`.

Reply:
You're right, what I meant to say was that adding the required attribute lets the browser check if the user has emptied the field and forgot to put a value back in.  I think the form behaves in a nicer way with the required attribute but I can revert those changes if you want

Reply:
By submitting this form with an empty value for made_in:
<img width="559" alt="Capture d’écran 2020-11-09 à 20 29 12" src="https://user-images.githubusercontent.com/1884246/98587332-49184e80-22ca-11eb-9b04-933142f5b76f.png">

Here is what I get and want to prevent by adding the required tag:
<img width="186" alt="Capture d’écran 2020-11-09 à 20 29 21" src="https://user-images.githubusercontent.com/1884246/98587327-487fb800-22ca-11eb-8535-a1213df52d4c.png">